src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,134 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+
 public class StringManipulation implements StringManipulationInterface {
+    String string = null;
+    // Default constructor
+    StringManipulation() {} 
+ 
+
+    // Parameterized constructor
+    StringManipulation(String string) {
```

Comment:
I see you have separate constructors, one default and other parameterized.  I was thinking you could merge and have one constructor.  What do you think?